extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/handler/AstroThingHandler.java

```diff
@@ -164,112 +187,121 @@ private void restartJobs() {
         if (schedulerFuture != null && !schedulerFuture.isCancelled()) {
             schedulerFuture.cancel(true);
         }
-
-        schedulerFuture = scheduler.schedule(new Runnable() {
-            @Override
-            public void run() {
+        if (localExecutor != null) {
+            localExecutor.shutdownNow();
+        }
+        localExecutor = Executors.newSingleThreadExecutor();
```

Comment:
Calling `run()` directly will block the thread that schedules all the jobs. We can do the following as well and leave the threads to the garbage collectors as soon as they finish.

```java
scheduler.schedule(() -> {
	.....
	.....
	new Thread().start(..);
	.....
	new Thread().start(..);
	.....
});
```

As the jobs are not using any time consuming resources, we still can create two threads to execute the jobs and leave it to the garbage collector for cleanup.  For manual cleanup, I have used the single thread executor.